File: Completed-Algorithms/Leetcode_56_Merge_Overlapping_Arrays/interval.js
/**
 * @param {number[][]} intervals
 * @return {number[][]}
 */
var merge = function(intervals) {
    // Sort the intervals by the start
    intervals.sort((a,b) => a[0] - b[0])
    const results  = []
    intervals.forEach(interval => {
        //if empty or the intervals first item is greater than the last item in existing array add
        if(results.length === 0 || interval[0] > results[results.length - 1][1]) {
            results.push(interval)
        } else {
            // othrwise, we will make the last end index of results the greater between the interval end and results end
            results[results.length - 1][1] = Math.max(results[results.length - 1][1], interval[1])
        }
    })
    return results
};

File: Completed-Algorithms/Leetcode_56_Merge_Overlapping_Arrays/interval.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort(key=lambda x: x[0])
        results = []
        for interval in intervals:
            if not results or interval[0] > results[-1][1]:
                results.append(interval)
            else:
                results[-1][1] = max(results[-1][1], interval[1])
        return results
